packages/openapi3/src/openapi.ts

```diff
@@ -458,6 +460,19 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
     }
   }
 
+  function getEffectiveType(type: Type): Type {
+    if (type.kind === "Model" && !type.name) {
+      const filtered = program.checker.filterModelProperties(type, isSchemaProperty);
+      if (filtered !== type) {
+        const effective = program.checker.getEffectiveModelType(filtered);
+        if (effective.name) {
+          return effective;
+        }
+      }
```

Comment:
I think you can go one more step here and check `type` if you didn't find a match for `effective`:
```suggestion
      }
      const effective = program.checker.getEffectiveModelType(type);
      if (effective.name) {
        return effective;
      }
```

Reply:
Do you have one or more examples of the cases you had in mind to address here?

I pushed a simplification that will handle degenerate case `op x(): {} & Data`, but I'm missing the case with `@header` involved where this would help.

Reply:
Hmm, so this now caused a test to fail:

```js
it("doesn't define anonymous or unconnected models", async () => {
```

Arguably, if someone does `{ ...Foo }` they are trying to inline Foo. So maybe this fix is too aggressive? Then again, if Foo had a header and you did `{...Foo}` it would not inline even without the recent push, so I'm not sure what to do here. Maybe we need another mechansim to say "please inline"?

Reply:
This is the case I am thinking of:
```
import "@cadl-lang/rest";

@serviceTitle("Widget Service")
namespace DemoService;
using Cadl.Http;

model Widget {
  @header etag: string;
  id: string;
}

interface WidgetService {
  @route("widgets/{id}") @get read(@path id: string): {@statusCode code: 200 } & Widget;
}
```

Reply:
Ah! Thanks.

Reply:
OK, I don't think this case is handled differently by that proposed change, though. Neither `type` nor `filtered` have an effective model type of `Widget`. `type` is not equivalent to `Widget` because it has extra field `statusCod`. `filtered` is not equivalent to `Widget` because it is missing field `etag`. To make this case work, I  think there has to be some extra information on the etag field that it is not to be filtered when looking for equivalent type. `@logical` or something.

Reply:
A common case here is likely 
```
model ServiceMetadataProps {
  @statusCode code: 200;
  @header serviceCommonHeader: string;
}

@route("widgets/{id}")  @get read(@path id: string): ServiceMetadataProps & Widget;
```

In this case, we would like the logical model to be `Widget`, but the schema in swagger should match `filtered`.  This *might* require a decorator like `@metadata` or `@logical` so the author can be absolutely clear about intentions.  You could imagine, for example, a logical response that consisted only of response headers, which would be impossible to differentiate from a common service metadata model.

Reply:
Working on a change that should make this work without needing extra decorators after discussing this with @bterlson.

Reply:
This case now works: https://cadlplayground.z22.web.core.windows.net/prs/506/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOw0KDQpAc2VydmljZVRpdGxlKCJXaWRnZXQgU8YVIikNCm5hbWVzcGFjZSBEZW1vxxk7DQp1c2luZyBDYWRsLkh0dHDFTW1vZGVsIMdEew0KICBpZDogc3RyaW5nOw0KfcQkaW50ZXJmxFDGKMdSxi9Acm91dGUoInfFG3Mve2lkfSIpIEDEUHJlYWQoQHBhdGjLVik6IHsgQHN0YXR1c0NvZGUgY29kZTogMjAwOyB9ICbHZMZ8

I'm going to add some function documentation and then this should be ready to go.